Failing tests

I encountered issue with running `go test` in the project's root. Various tests failed for me.

**Example:**

`traveler_unit_test.go`, function `TestCarbon_SubNanoseconds`:

I expected the test to pass, but it didn't:

```
Running tool: /usr/local/go/bin/go test -timeout 30s -run ^TestCarbon_SubNanoseconds$ github.com/dromara/carbon/v2

--- FAIL: TestCarbon_SubNanoseconds (0.00s)
    --- FAIL: TestCarbon_SubNanoseconds/case2 (0.00s)
        /home/andreas/carbon/traveler_unit_test.go:2925: 
            	Error Trace:	/home/andreas/carbon/traveler_unit_test.go:2925
            	Error:      	Not equal: 
            	            	expected: "2020-08-05 13:14:15.222222221 +0800 CST"
            	            	actual  : "2020-08-05 13:14:15.222222221 +0200 CEST"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-2020-08-05 13:14:15.222222221 +0800 CST
            	            	+2020-08-05 13:14:15.222222221 +0200 CEST
            	Test:       	TestCarbon_SubNanoseconds/case2
            	Messages:   	SubNanoseconds()
    --- FAIL: TestCarbon_SubNanoseconds/case3 (0.00s)
        /home/andreas/carbon/traveler_unit_test.go:2925: 
            	Error Trace:	/home/andreas/carbon/traveler_unit_test.go:2925
            	Error:      	Not equal: 
            	            	expected: "2020-08-05 13:14:15.22222222 +0800 CST"
            	            	actual  : "2020-08-05 13:14:15.22222222 +0200 CEST"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-2020-08-05 13:14:15.22222222 +0800 CST
            	            	+2020-08-05 13:14:15.22222222 +0200 CEST
            	Test:       	TestCarbon_SubNanoseconds/case3
            	Messages:   	SubNanoseconds()
FAIL
FAIL	github.com/dromara/carbon/v2	0.006s
FAIL
```

I suspect the reason is the fact that I am on a time zone different from the test's author. My time zone:

```go
$ timedatectl
               Local time: Do 2024-12-12 21:45:43 CET
           Universal time: Do 2024-12-12 20:45:43 UTC
                 RTC time: Do 2024-12-12 21:45:44
                Time zone: Europe/Berlin (CET, +0100)
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -2,8 +2,6 @@ name: test
 
 on:
   push
-env:
-  TZ: "PRC"
 jobs:
 
   test:

--- a/database_unit_test.go
+++ b/database_unit_test.go
@@ -913,6 +913,8 @@ func TestCarbon_Issue240(t *testing.T) {
 
 // https://github.com/dromara/carbon/issues/243
 func TestCarbon_Issue243(t *testing.T) {
+	prepareTest(t)
+
 	SetDefault(Default{
 		Layout:       DateTimeLayout,
 		Timezone:     PRC,

--- a/default_bench_test.go
+++ b/default_bench_test.go
@@ -3,6 +3,8 @@ package carbon
 import "testing"
 
 func BenchmarkCarbon_SetDefault(b *testing.B) {
+	prepareTest(b)
+
 	d := Default{
 		Layout:       DateTimeLayout,
 		Timezone:     Local,

--- a/default_unit_test.go
+++ b/default_unit_test.go
@@ -7,6 +7,8 @@ import (
 )
 
 func TestCarbon_SetDefault(t *testing.T) {
+	prepareTest(t)
+
 	SetDefault(Default{
 		Layout:       DateTimeLayout,
 		Timezone:     PRC,

--- a/test.go
+++ b/test.go
@@ -1,6 +1,7 @@
 package carbon
 
 import (
+	"os"
 	"testing"
 )
 
@@ -20,6 +21,18 @@ func (c *Carbon) UnSetTestNow() {
 // 是否设置过当前测试时间
 func (c Carbon) IsSetTestNow() bool {
 	return c.testNow > 0
+}
+
+// TestMain sets up the testing environment for all tests
+// https://pkg.go.dev/testing#hdr-Main
+func TestMain(m *testing.M) {
+	// The whole tests were written for PRC timezone (China).
+	// The codebase of test is too large to be changed.
+	// Without this hack the tests will fail if you use a different timezone than PRC
+	// This will affect the way Go compute the timezone when using time.Local
+	_ = os.Setenv("TZ", "PRC")
+
+	m.Run()
 }
 
 func prepareTest(tb testing.TB) {

--- a/test.go
+++ b/test.go
@@ -1,5 +1,9 @@
 package carbon
 
+import (
+	"testing"
+)
+
 // SetTestNow sets a test Carbon instance (real or mock) to be returned when a "now" instance is created.
 // 设置当前测试时间
 func (c *Carbon) SetTestNow(carbon Carbon) {
@@ -16,4 +20,20 @@ func (c *Carbon) UnSetTestNow() {
 // 是否设置过当前测试时间
 func (c Carbon) IsSetTestNow() bool {
 	return c.testNow > 0
+}
+
+func prepareTest(tb testing.TB) {
+	tb.Helper()
+
+	// Store the current default
+	savedDefault := Default{
+		Layout:       defaultLayout,
+		Timezone:     defaultTimezone,
+		Locale:       defaultLocale,
+		WeekStartsAt: defaultWeekStartsAt,
+	}
+	tb.Cleanup(func() {
+		// restore the default when test is done
+		SetDefault(savedDefault)
+	})
 }

